app/src/main/java/com/hamdan/forzenbook/view/composeutils/Composables.kt

```diff
@@ -156,36 +197,61 @@ fun LoginBackgroundColumn(modifier: Modifier = Modifier, content: @Composable ()
 
 @Composable
 fun LoginTopBar(topText: String) {
-    TopAppBar(
-        modifier = Modifier.fillMaxWidth(),
-        backgroundColor = ForzenbookTheme.colors.colors.background,
-    ) {
-        val navController = LocalNavController.current
-        Column(modifier = Modifier.weight(1f), verticalArrangement = Arrangement.Center) {
+    val navController = LocalNavController.current
+    val navigationIcon: (@Composable () -> Unit) = {
+        IconButton(onClick = { navController?.navigateUp() }) {
             Icon(
-                painter = painterResource(id = R.drawable.ic_round_arrow_back_ios_24),
+                imageVector = Icons.Filled.ArrowBack,
                 contentDescription = stringResource(id = R.string.back_arrow),
-                modifier = Modifier
-                    .size(IconSizeValues.small_2)
-                    .clickable {
-                        navController?.navigateUp()
-                    },
-                tint = ForzenbookTheme.colors.colors.onBackground
+                tint = ForzenbookTheme.colors.colors.onBackground,
             )
         }
-        Column(
-            modifier = Modifier.weight(1f),
-            verticalArrangement = Arrangement.Center,
-            horizontalAlignment = Alignment.CenterHorizontally
-        ) {
+    }
+    TopAppBar(
+        title = {
             Text(
                 text = topText,
-                fontSize = ForzenbookTheme.typography.h3.fontSize,
+                fontSize = ForzenbookTheme.typography.h2.fontSize,
                 textAlign = TextAlign.Center,
-                color = ForzenbookTheme.colors.colors.onBackground
+                color = ForzenbookTheme.colors.colors.onBackground,
+                maxLines = ONE_LINE,
+                overflow = TextOverflow.Ellipsis,
+            )
+        },
+        modifier = Modifier.fillMaxWidth(),
+        backgroundColor = ForzenbookTheme.colors.colors.background,
+        navigationIcon = navigationIcon,
+        actions = {
+            Icon(
```

Comment:
Are you sure this is being used right? Two back icons?

Reply:
yes one is an invisible icon to take up space so the createaccount on small screens ellipsizes on screen instead of edge of screen, basicaly so the text stays centered.

Reply:
I was looking into ways to make the bar have inner padding on the right and wasn't seeing any.